app/src/main/java/net/grandcentrix/blutufintegration/ui/list/ListViewModel.kt

```diff
@@ -55,7 +55,9 @@ class ListViewModel(
     fun startScan() {
         viewModelScope.launch {
             _scanState.value = true
-            scanUseCase.execute()
+            Transformations.map(scanUseCase.execute().asLiveData()) { resource ->
+                uiModel.value = resource
+            }
```

Comment:
I think we should use this here instead of misusing a mapping for collecting items.
```suggestion
            scanUseCase.execute().collect { resource ->
                uiModel.value = resource
            }
```

Reply:
👍👍